### PR TITLE
feat(context): introduce BrowserContext close event

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: docs/api.md [hunk 1]
```diff
@@ -274,6 +275,13 @@ await context.close();
 - [browserContext.setPermissions(origin, permissions[])](#browsercontextsetpermissionsorigin-permissions)
 <!-- GEN:stop -->
 
+#### event: 'close'
+
+Emitted when Browser context gets closed. This might happen because of one of the following:
+- Browser context is closed.
```
Comment (JoelEinbinder) [inline docs/api.md]: Do we emit page close on disconnect? Maybe we should? I think we should be consistent.

Comment (pavelfeldman) [reply docs/api.md]: Should, we can do that.

Comment (pavelfeldman) [reply docs/api.md]: @JoelEinbinder turns out it is fired. Added a test.

File: test/launcher.spec.js [hunk 2]
```diff
@@ -169,6 +169,19 @@ module.exports.describe = function({testRunner, expect, defaultBrowserOptions, p
...
+      const remote = await playwright.connect({ wsEndpoint: browserServer.wsEndpoint() });
+      const context = await remote.newContext();
+      const page = await context.newPage();
+      let contextClosed = false;
+      let pageClosed = false;
+      context.on('close', e => contextClosed = true);
+      page.on('close', e => pageClosed = true);
+      await browserServer.close();
+      expect(contextClosed).toBeTruthy();
+      expect(pageClosed).toBeTruthy();
```
Comment (dgozman) [inline test/launcher.spec.js]: I believe if we call `remote.close()`, we will get `contextClosed`, but not `pageClosed`. Let's unify?

File: docs/api.md [hunk 3]
```diff
@@ -274,6 +275,13 @@ await context.close();
...
+- Browser application is closed or crashed.
+- The [`browser.disconnect`](#browserdisconnect) method was called.
```
Comment (dgozman) [inline docs/api.md]: There is no such method anymore, it's called `browser.close` now.

**Review by dgozman — APPROVED** [review]

File: test/launcher.spec.js (hunk 2 above)
Comment (pavelfeldman) [reply test/launcher.spec.js]: I believe we issue pageclosed.

File: docs/api.md (hunk 3 above)
Comment (pavelfeldman) [reply docs/api.md]: Done

**Pull request merged**